File: perl/src/tests/t_create_slave.pl
#!/usr/bin/perl
#
# $Id:$
#
# perl test driver

use Atrshmlog;

$result = Atrshmlog::attach();

Atrshmlog::sleep_nanos(1000000000);

$akcount = 0;

$slave = undef;

$slave = Atrshmlog::get_next_slave_local($slave);

while ($slave) {
    ++$akcount;

    $slave = Atrshmlog::get_next_slave_local($slave);
}

print "create slave : $akcount : \n";

$s = Atrshmlog::create_slave();

Atrshmlog::sleep_nanos(1000000000);

print "create slave : $s : \n";

$akcount = 0;

$slave = undef;

$slave = Atrshmlog::get_next_slave_local($slave);

while ($slave) {
    ++$akcount;

    $slave = Atrshmlog::get_next_slave_local($slave);
}

print "create slave : $akcount : \n";


print " \n";

exit (0);

# end of main

File: python/src/tests/t_create_slave.py
#!/usr/bin/python3
#
# $Id:$
#
# We test a bit of the atrshmlog here.
#
# This is for the first starter, so only the basic things.

import sys
import atrshmlog

r = atrshmlog.attach()

atrshmlog.sleep_nanos(1000000000)

akcount = 0

slave = atrshmlog.get_next_slave_local(0)

while slave != 0:
    akcount = akcount + 1
    slave = atrshmlog.get_next_slave_local(slave)


print('create slave : ' + str(akcount) + ' : ')

s = atrshmlog.create_slave()

atrshmlog.sleep_nanos(1000000000)

print('create slave : ' + str(s) + ' : ')

akcount = 0

slave = atrshmlog.get_next_slave_local(0)

while slave != 0:
    akcount = akcount + 1
    slave = atrshmlog.get_next_slave_local(slave)


print('create slave : ' + str(akcount) + ' : ')


print (' ')

exit(0);

# end of test

